src/components/BaseButton.vue

```diff
@@ -2,18 +2,20 @@
   <button
     class="inline-flex items-center font-medium text-white rounded-md shadow-sm"
     :class="{
-      'cursor-not-allowed': disabled,
+      'cursor-not-allowed': disabled || loading,
```

Comment:
This code `disabled || loading` is repeating 4 times, you can write a computed method for this